src/common

```diff
@@ -229,23 +229,16 @@ OBJ += $(patsubst %.cc, %$(OBJEXT), $(filter %.cc, $(SRC)))
 
 # get version from git
 GIT_INFO = $(shell git describe --tags --always --dirty || echo "n/a")
-GIT_INFO_FILE = .release_info
+RELEASE_INFO = "\#define CBMC_VERSION \"$(CBMC_VERSION) ($(GIT_INFO))\""
+GIT_INFO_FILE = version.h
 
-CBMC_VERSION_FILES += $(filter %_parse_options$(OBJEXT), $(OBJ))
+$(GIT_INFO_FILE):
+	echo $(RELEASE_INFO) > $@
 
-ifeq ($(BUILD_ENV_),MSVC)
-$(CBMC_VERSION_FILES): CP_CXXFLAGS += /DCBMC_VERSION='"""$(CBMC_VERSION) ($(GIT_INFO))"""'
-else
-$(CBMC_VERSION_FILES): CP_CXXFLAGS += -DCBMC_VERSION="\"$(CBMC_VERSION) ($(GIT_INFO))\""
-endif
+$(filter %_parse_options$(OBJEXT), $(OBJ)): $(GIT_INFO_FILE)
```

Comment:
Given that the version info is now pulled in by a standard `#include` - do you actually need to add explicit rules for the dependency any more? Wouldn't that be picked up by `.d` files just like any other include ?

Reply:
I had thought the same, but actually this isn't true: the `.d` file will only be built as a side-effect of compilation, and thus only helps for re-builds - it won't do the trick in a clean build tree!

Reply:
Oh right, I see. I wonder if we can arrange for the generation target to be run in the first build without having to add these dependencies explicitly like this...